### PR TITLE
refactor: Leverage Multimaps 

### LINKED ISSUE #745: Leverage Multimaps
You do not need GtfsTripTableContainer since you already have GtfsStopTimeTableContainer. You can make a cycle over `Multimaps.asMap(table.byTripIdMap())`.

_Originally posted by @aababilov in https://github.com/MobilityData/gtfs-validator/pull/699#discussion_r575130942_

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #745 

**Summary:**

This PR improves code quality by leveraging `Multimaps`

**Expected behavior:** 

Same tests should pass.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Review by barbeau — APPROVED** [review]
👍 , thanks @lionel-nj!

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MissingTripEdgeValidator.java [hunk 1]
```diff
@@ -33,16 +32,13 @@
...
-    tripTable
-        .getEntities()
+    Multimaps.asMap(stopTimeTable.byTripIdMap())
         .forEach(
-            trip -> {
-              List<GtfsStopTime> stopTimesForTrip = stopTimeTable.byTripId(trip.tripId());
+            (tripId, stopTimesForTrip) -> {
               if (stopTimesForTrip.isEmpty()) {
                 return;
               }
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/MissingTripEdgeValidator.java:44]: This check is redundant. stopTimesForTrip may never be empty here.

**Comment by aababilov** [comment]
Thanks for doing this! Do you mind to wait for my review next time?

**Comment by lionel-nj** [comment]
> Thanks for doing this! Do you mind to wait for my review next time?

Sure!

**Pull request merged**